fog/ledger/enclave/impl/src/lib.rs

```diff
@@ -252,41 +265,30 @@ where
             .into_iter()
             .map(|(responder_id, query_response)| {
                 let plaintext_bytes = self.ake.backend_decrypt(&responder_id, &query_response)?;
-                let query_response: CheckKeyImagesResponse =
-                    mc_util_serial::decode(&plaintext_bytes)?;
+                let query_response: ShardKeyImageResponse =
+                    mc_util_serial::deserialize(&plaintext_bytes)?;
 
                 Ok(query_response)
             })
-            .collect::<Result<Vec<CheckKeyImagesResponse>>>()?;
+            .collect::<Result<Vec<_>>>()?;
```

Comment:
I'm wasn't familiar with this syntax - after a quick search, it looks like the compiler will infer the vector type, which iiuc will be `Vec<EnclaveMessage<NonceSession>>`

Reply:
it will be a `Vec<ShardKeyImageResponse>`, image from rust rover, 
![image](https://github.com/mobilecoinfoundation/mobilecoin/assets/101757619/97b3ca99-8e5f-414a-8b52-f5b45f331fd7)

I chose to use type inference here as I changed the type of `query_response` and it had to be changed in multiple places so decided to use inference to minimize future maintenance.
I will admit type inference is something newer to me coming from C, so it may be a hammer I swing a bit too much.